rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -257,18 +260,15 @@ public RepositoryLinksResource process(RepositoryLinksResource resource) {
         HalResource<Project> halProject = new HalResource<>(sw360Project);
 
         restControllerHelper.addEmbeddedUser(halProject, sw360User, "createdBy");
-        Set<String> releaseIds = new HashSet<>();
-        if (sw360Project.getReleaseIdToUsage() != null) {
-            Map<String, ProjectReleaseRelationship> releaseIdToUsage = sw360Project.getReleaseIdToUsage();
-            for (String releaseId : releaseIdToUsage.keySet()) {
-                if (releaseIdToUsage.get(releaseId).releaseRelation.equals(ReleaseRelationship.CONTAINED)) {
-                    releaseIds.add(releaseId);
-                }
-            }
-            restControllerHelper.addEmbeddedReleases(halProject, releaseIds, releaseService, sw360User, "containedReleases");
-            sw360Project.setReleaseIdToUsage(null);
+        Map<String, ProjectReleaseRelationship> releaseIdToUsage = sw360Project.getReleaseIdToUsage();
+        if (!Objects.isNull(releaseIdToUsage)) {
+            restControllerHelper.addEmbeddedReleases(halProject, releaseIdToUsage.keySet(), releaseService, sw360User);
+        }
+        Map<String, ProjectRelationship> linkedProjects = sw360Project.getLinkedProjects();
+        if (!Objects.isNull(linkedProjects)) {
+            restControllerHelper.addEmbeddedProject(halProject, linkedProjects.keySet(), projectService, sw360User);
         }
-        if (sw360Project.getModerators() != null) {
+        if (!Objects.isNull(sw360Project.getModerators())) {
```

Comment:
Why did you change a perfectly readable code to this contortion? (Here and above as well)

Quote from the javadoc:

> This method exists to be used as a Predicate, filter(Objects::isNull)

And then there's the method `Objects.nonNull()`... 

I really don't see the point. You're not making the code any more readable and you're misusing the helper method.

Reply:
stupid habit... corrected to !=null